### PR TITLE
added express & supertest, set up express server with first endpoint,…

### PR DESCRIPTION
**Pull request by sinnedchan**
… added logic for GET /api/topics with happy and sad path testing

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    it("array should be length of 3", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.topics.length).toBe(3);
+        });
+    });
+    it("each topic should be an object with 2 keys (slug + description), both having strings", () => {
+      return request(app)
```
Comment (popatre) [inline __tests__/app.test.js:35]: You could add all the expect statement above into a single test here. Your test suite can have multiple expects.

File: __tests__/errors.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const app = require("../app");
+
+describe("global errors", () => {
+  describe("404 endpoint not found", () => {
+    it("should return a status code of 404 if trying to access and endpoint that doesn't exist", () => {
+      return request(app).get("/invalidpath").expect(404);
+    });
+    it("return an object with a status code and a message", () => {
+      return request(app)
+        .get("/invalidpath")
```
Comment (popatre) [inline __tests__/errors.test.js]: This test basically covers the one above, so dont need the test above.

**Review by popatre — APPROVED** [review]
Remove the redundant test 
Could also add your happy path tests into a single test, as they're all checking the same happy path endpoint.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -9,35 +9,40 @@ afterAll(() => db.end());
...
-    it("responds with status code of 200", () => {
-      return request(app).get("/api/topics").expect(200);
-    });
-    it("reponds with an object containing an array", () => {
+    it("responds with a status code of 200 and an object containing an array with a length of 3, each topic should be an object with 2 keys (slug + description), both having strings ", () => {
       return request(app)
         .get("/api/topics")
         .expect(200)
         .then(({ body }) => {
+          expect(200);
```
Comment (xViolaine) [inline __tests__/app.test.js]: You are already expecting 200 on line 15 - this is redundant

**Review by xViolaine — APPROVED** [review]

**Pull request merged**